plugins/modules/aci_cloud_external_epg_selector.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Anvitha Jain (@anvitha-jain) <anvjain@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: aci_cloud_external_epg_selector
+short_description: Manage Cloud Endpoint Selector for External EPGs (cloud:ExtEPSelector)
+description:
+- Decides which endpoints belong to the EPGs based on several parameters.
+notes:
+- More information about the internal APIC class B(cloud:ExtEPSelector) from
+  L(the APIC Management Information Model reference,https://developer.cisco.com/docs/apic-mim-ref/).
+author:
+- Anvitha Jain (@anvitha-jain)
+options:
+  name:
+    description:
+    - The name of the Cloud Endpoint selector.
+    aliases: [ selector, cloud_external_epg_selector, external_epg_selector, extepg_selector ]
+    type: str
+  subnet:
+    description:
+    - Ip address of the Cloud Subnet.
+    type: str
+  tenant:
+    description:
+    - The name of tenant.
+    type: str
+  ap:
+    description:
+    - The name of the cloud application profile.
+    aliases: [ app_profile, app_profile_name ]
+    type: str
+  cloud_external_epg:
+    description:
+    - Name of Object cloud_external_epg.
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    default: present
+    type: str
+extends_documentation_fragment:
+- cisco.aci.aci
+'''
+
+EXAMPLES = r'''
+- name: Add a new cloud external EPG selector
+  cisco.aci.aci_cloud_external_epg_selector:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: tenant1
+    ap: ap1
+    cloud_external_epg: ext_epg
+    name: subnet_name
+    subnet: 10.0.0.0/16
+    state: present
+  delegate_to: localhost
+
+- name: Remove a cloud external EPG selector
+  cisco.aci.aci_cloud_external_epg_selector:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    validate_certs: no
+    tenant: tenant1
+    ap: ap1
+    cloud_external_epg: ext_epg
+    name: subnet_name
+    subnet: 10.0.0.0/16
+    state: absent
+  delegate_to: localhost
+
+- name: Query a cloud external EPG selector
```

Comment:
Can we add query all if applicable?

Reply:
only query all is available. query specific isn't